version: '3.8'
services:
  fastapi:
    build:
      context: .
      dockerfile: ./Dockerfile
    command: "uvicorn main:app --host 0.0.0.0 --port 8000"
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - myapp:/app
    restart: always
    environment:
      - DATABASE_URL=postgresql://${DB_USER}:${DB_PASSWORD}@db:5432/${DB_NAME}?schema=public
    env_file:
      - .env
    networks:
      - postgres-db-network

  db:
    image: postgres:15
    container_name: postgres
    environment:
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
      - mainproject_postgres_data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    restart: always
    networks:
      - postgres-db-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER}"]
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  mainproject_postgres_data:
    name: mainproject_postgres_data
  myapp:

networks:
    postgres-db-network:
        driver: bridge